data_steward/analytics/cdr_ops/rdr_export_qc.py

```diff
@@ -0,0 +1,308 @@
+
+# ## Concept codes used
+# Identify question and answer concept codes which were either added or removed (appear in only the new or only the old RDR datasets, respectively).
+
+query = f'''
+WITH curr_code AS (
+SELECT observation_source_value value, 
+ 'observation_source_value' field, 
+ COUNT(1) row_count 
+FROM {project_id}.{new_rdr}.observation GROUP BY 1
+
+UNION ALL
+
+SELECT value_source_value value, 
+ 'value_source_value' field, 
+ COUNT(1) row_count 
+FROM {project_id}.{new_rdr}.observation GROUP BY 1),
+
+prev_code AS (
+SELECT observation_source_value value,
+ 'observation_source_value' field, 
+ COUNT(1) row_count 
+FROM {project_id}.{old_rdr}.observation GROUP BY 1
+
+UNION ALL
+
+SELECT value_source_value value, 
+ 'value_source_value' field, 
+ COUNT(1) row_count 
+FROM {project_id}.{old_rdr}.observation GROUP BY 1)
+
+SELECT 
+  prev_code.value prev_code_value, 
+  prev_code.field prev_code_field, 
+  prev_code.row_count prev_code_row_count, 
+  curr_code.value curr_code_value, 
+  curr_code.field curr_code_field, 
+  curr_code.row_count curr_code_row_count, 
+FROM curr_code 
+ FULL OUTER JOIN prev_code
+  USING (field, value)
+WHERE prev_code.value IS NULL OR curr_code.value IS NULL
+'''
+pd.read_gbq(query, dialect='standard')
+
+# # Check if observation_source_value vs concept ids
+
+query = f"""
+SELECT 
+    SUM(CASE WHEN observation_source_concept_id IS NULL THEN 1 ELSE 0 END) as n_null_source_concept_id
+    , SUM(CASE WHEN observation_source_concept_id=0 THEN 1 ELSE 0 END) as n_zero_source_concept_id
+FROM `{project_id}.{new_rdr}.observation`
+WHERE observation_source_value IS NOT NULL
+and observation_source_value != ''
+"""
+pd.read_gbq(query,
+            dialect='standard')
+
+query = f"""
+SELECT 
+    SUM(CASE WHEN observation_concept_id IS NULL THEN 1 ELSE 0 END) as n_null_concept_id
+    , SUM(CASE WHEN observation_concept_id=0 THEN 1 ELSE 0 END) as n_zero_concept_id
+FROM `{project_id}.{new_rdr}.observation`
+WHERE observation_source_value IS NOT NULL
+and observation_source_value != ''
+"""
+pd.read_gbq(query,
+            dialect='standard')
+
+# # Check value_source_value unmapped to source
+
+query = f"""
+SELECT 
+    SUM(CASE WHEN value_source_concept_id IS NULL THEN 1 ELSE 0 END) as n_null_source_concept_id
+    , SUM(CASE WHEN value_source_concept_id=0 THEN 1 ELSE 0 END) as n_zero_source_concept_id
+FROM `{project_id}.{new_rdr}.observation`
+WHERE value_source_value IS NOT NULL
+and value_source_value != ''
+"""
+pd.read_gbq(query,
+            dialect='standard')
+
+# # Check value_source_value unmapped to standard
+
+query = f"""
+SELECT 
+    SUM(CASE WHEN value_as_concept_id IS NULL THEN 1 ELSE 0 END) as n_null_concept_id
+    , SUM(CASE WHEN value_as_concept_id=0 THEN 1 ELSE 0 END) as n_zero_concept_id
+FROM `{project_id}.{new_rdr}.observation`
+WHERE value_source_value IS NOT NULL
+and value_source_value != ''
+"""
+pd.read_gbq(query,
+            dialect='standard')
```

Comment:
You could do one of two things here.
1. Just add these SUM/CASE statements to the previous query.
2. Parameterize the previous query and run it twice with different field values.

Reply:
Good idea- I did 1. Also I added the value_source_value so we know which codes have the problem.